model/general.yaml

```
@@ -34,15 +34,16 @@ groups:
       - ref: peer.service
         requirement_level: recommended
   - id: identity
-    type: span
+    type: attribute_group
     brief: >
-      These attributes may be used for any operation with an authenticated and/or authorized enduser.
+      These attributes may be used for any operation with an authenticated and/or authorized user.
     attributes:
-      - ref: enduser.id
-        requirement_level: recommended
-      - ref: enduser.role
+      - ref: user.id
+        requirement_level:
+          conditionally_required: If instrumentation supports tracking unauthenticated users and if `user.authentication.id` is set, recommended otherwise.
```

sorry, I think I got it wrong in my previous suggestion.

We don't want `user.id` to be used for authenticated users, so I guess it should be

```suggestion
          conditionally_required: If and only if instrumentation supports tracking unauthenticated users and if `user.authentication.id` is set.
```

wdyt?

---

the goal is to prevent someone from duplicating things like 

❌ Bad:

```
user.authentication.id = lmolkova
user.id = lmolkova
```

❌ Bad:

```
user.id = lmolkova
```

we want people to do 

✅  Good:

```
user.id = QdH5CAWJgqVT4rOr0qtumf
```

✅  Good:

```
user.id = QdH5CAWJgqVT4rOr0qtumf
user.authentication.id = lmolkova
```

✅  Good:

```
user.authentication.id = lmolkova
```

---

Maybe another way to same this is that this value SHOULD NOT be identifying in anyway to the real user id. So this value MUST not contain PII which can directly identify the authenticated user.

---

PII is once concern, but my main point is is avoid duplication (you should not generate new guid for the sake of populating user.id) and consistency - everyone should be using `user.*.id` in the same way

---

for azure monitor javascript SDK, browser always has a cookie named 'ai_user=QdH5CAWJgqVT4rOr0qtumf`. this will be tracked via `user.id` regardless of whether user is authenticated or not. user can choose to use `user.authentication.id` for the real user's id/name.

can we make it not required and remove the condition completely?

---

I'm missing something. 

The condition here is if user tracking is supported, so azmon distro should set `user.id` based on it.

this condition says nothing about authentication.

---

all i'm saying that `user.id` is not a required field.

---

would this work?

```suggestion
          recommended: If and only if instrumentation supports tracking unauthenticated users.
```

---

> i'm saying that user.id is not a required field.

It might be not required for this specific use case. But for anything related to the users in OS `user.id` is well known and important field